raft/broadcast_tables/experimental/lang.cc

```diff
@@ -22,7 +24,95 @@ bool is_broadcast_table_statement(const sstring& keyspace, const sstring& column
 }
 
 future<> execute(service::raft_group0_client& group0_client, const query& query) {
-    throw exceptions::invalid_request_exception{"executing queries on broadcast_kv_store is currently not implemented"};
+    auto group0_cmd = group0_client.prepare_command(service::broadcast_table_query{query});
+    return group0_client.add_entry_unguarded(std::move(group0_cmd));
+}
+
+
+static
+std::pair<lw_shared_ptr<::query::read_command>, dht::partition_range>
+prepare_read_command(service::storage_proxy& proxy, const schema& schema, const bytes& key) {
+    auto slice = partition_slice_builder(schema).build();
+    auto partition_key = partition_key::from_single_value(schema, key);
+    dht::ring_position ring_position(dht::get_token(schema, partition_key), partition_key);
+    auto range = dht::partition_range::make_singular(ring_position);
+    return {make_lw_shared<::query::read_command>(
+            schema.id(),
+            schema.version(),
+            slice,
+            proxy.get_max_result_size(slice),
+            ::query::tombstone_limit(proxy.get_tombstone_limit())
+        ), range};
+}
+
+static
+const atomic_cell_view
+get_atomic_cell(const schema_ptr& schema, mutation& mutation, const bytes& name) {
+    const auto* column_definition = schema->get_column_definition(name);
+    return mutation.partition().clustered_row(*schema, clustering_key::make_empty()).cells().cell_at(column_definition->id).as_atomic_cell(*column_definition);
+}
+
+future<> execute_broadcast_table_query(
+    service::storage_proxy& proxy,
+    const query& query,
+    utils::UUID cmd_id) {
+    return std::visit(make_visitor (
+    std::bind_front([] (service::storage_proxy& proxy, const raft::broadcast_tables::select_query& q) -> future<> {
```

Comment:
Need some indents here.

Reply:
We usually don't indent the arguments to `make_visitor`, from a quick glance through the codebase

Reply:
Well it looks bad. std::bind_front is a continuation of the return statement, without indents it will look like a separate thing to a casual reader.

Reply:
Added indent.